port/common/omrsysinfo_helpers.c

```diff
@@ -435,6 +435,35 @@ omrsysinfo_get_x86_description(struct OMRPortLibrary *portLibrary, OMRProcessorD
 	desc->features[3] = CPUInfo[CPUID_EBX]; /* Structured Extended Feature Flags in EBX */
 	desc->features[4] = CPUInfo[CPUID_ECX]; /* Structured Extended Feature Flags in ECX */
 
+	if (portLibrary->sysinfo_processor_has_feature(portLibrary, desc, OMR_FEATURE_X86_OSXSAVE)) {
```

Comment:
adding more code to `omrsysinfo_get_x86_description` bloats it, increases complexity and makes it harder to maintain. your previous approach of adding a new helper function, similar to `omrsysinfo_get_x86_cpuid_ext`, was a cleaner and modular approach. the helper should be re-introduced.

the helper should return a bitset. it can be used to directly set `desc->features[2]` with a bitwise-OR in line 432
```
	/* Mention which bits are still free. */
	desc->features[2] |= new_helper(...);
```